plaidml/edsl/edsl_docs.py

```diff
@@ -15,44 +15,38 @@ def sum_over_axis(I):
     m, n = TensorIndexes(2)
     I.bind_dims(M, N)
     O = TensorOutput(N)
-    O[n] += I[m, n]  # contraction
+    O = Contraction().outShape(N).outAccess(n).sum(I[m, n])
     return O
 
 
 def max_over_axis(I):
     M, N = TensorDims(2)
     m, n = TensorIndexes(2)
     I.bind_dims(M, N)
-    O = TensorOutput(N)
-    O[n] >= I[m, n]
-    return O
+    return Contraction().outShape(N).outAccess(n).max(I[m, n])
 
 
 def matmul(A, B):
     I, J, K = TensorDims(3)
     i, j, k = TensorIndexes(3)
     A.bind_dims(I, K)
     B.bind_dims(K, J)
-    C = TensorOutput(I, J)
-    C[i, j] += A[i, k] * B[k, j]
-    return C
+    return Contraction().outShape(I, J).outAccess(i, j).sum(A[i, k] * B[k, j])
 
 
 def global_min(I):
     i, j, k = TensorIndexes(3)
     Neg = -I
-    O_Neg = TensorOutput()
-    O_Neg[()] >= Neg[i, j, k]
-    O = -O_Neg
-    return O
+    O_Neg = Contraction().max(Neg[i, j, k])
+    return -O_Neg
```

Comment:
Does it turn out that we do need the separate `O_Neg` line for this to work correctly?

Reply:
There was a small issue with cpp. Please see the update.

Reply:
great, looks good!